pkg/controller/vsphere/reconciler_test.go

```diff
@@ -899,7 +899,7 @@ func TestDelete(t *testing.T) {
 	password, _ := server.URL.User.Password()
 	namespace := "test"
 	vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
-	instanceUUID := "instanceUUID"
+	instanceUUID := "a5764857-ae35-34dc-8f25-a9c9e73aa898"
```

Comment:
Was there not an extra test case here with an invalid UUID to test the behaviour of the fallback? Seems to have gone?

Reply:
Do you mean we should let invalid value stay here? Tests are passing in both cases anyway.

Reply:
I was wondering if there are any tests that actually cover the fallback to name? For that we would need an invalid ID right? But I couldn't see any tests where the ID was invalid, are there tests where the parse of the ID fails?

Reply:
What about this test https://github.com/openshift/machine-api-operator/pull/558/files#diff-1cda1db15f2622f564de1259e970d48eR161 ?

Reply:
I am blind 🤦  Ignore me :D